Assignment4/GNajeral-180275/task07.py

```diff
@@ -0,0 +1,71 @@
+# -*- coding: utf-8 -*-
+"""Task07.ipynb
+
+Automatically generated by Colaboratory.
+
+Original file is located at
+    https://colab.research.google.com/drive/1fzywZ4EwNykCSoc5kjbwf7_KAb8NKSC-
+
+**Task 07: Querying RDF(s)**
+"""
+
+!pip install rdflib 
+github_storage = "https://raw.githubusercontent.com/FacultadInformatica-LinkedData/Curso2021-2022/master/Assignment4/course_materials"
+
+"""Leemos el fichero RDF de la forma que lo hemos venido haciendo"""
+
+from rdflib import Graph, Namespace, Literal
+from rdflib.namespace import RDF, RDFS
+g = Graph()
+g.namespace_manager.bind('ns', Namespace("http://somewhere#"), override=False)
+g.namespace_manager.bind('vcard', Namespace("http://www.w3.org/2001/vcard-rdf/3.0#"), override=False)
+g.parse(github_storage+"/rdf/example6.rdf", format="xml")
+
+"""**TASK 7.1: List all subclasses of "Person" with RDFLib and SPARQL**"""
+
+from rdflib.plugins.sparql import prepareQuery
+NS = Namespace("http://somewhere#")
+q = prepareQuery('''
+   SELECT ?x WHERE { 
+     ?x rdfs:subClassOf ns:Person. 
+   }
+   ''',
+   initNs = { "ns": NS}
+ )
+for r in g.query(q):
+  print(r.x)
+
+"""**TASK 7.2: List all individuals of "Person" with RDFLib and SPARQL (remember the subClasses)**
+
+"""
+
+from rdflib.plugins.sparql import prepareQuery
+NS = Namespace("http://somewhere#")
+q = prepareQuery('''
+   SELECT ?x WHERE { 
+     {?x a ns:Person}
+     UNION
+     {?x a ns:Researcher}
```

Comment:
This is not correct, because if you had more subclasses of Person, you would not want filling them all manually. You should create a query to give you all the subclasses of Person to list individuals